### PR TITLE
Upgrade react native to 059

### PR DESCRIPTION
**Pull request by v-almonacid**

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -144,12 +148,12 @@ android {
     // Specifies one flavor dimension.
     flavorDimensions "version"
     productFlavors {
-        main {
+        main2 {
```
Comment (nicarq) [inline android/app/build.gradle]: @v-almonacid why this change?

File: babel.config.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-{
-  "presets": ["module:metro-react-native-babel-preset"],
+module.exports = {
+  presets: ['module:metro-react-native-babel-preset'],
```
Comment (nicarq) [inline babel.config.js]: I would use `"` because below we also use `"`

File: ios/emurgo/AppDelegate.m [hunk 3]
```diff
@@ -16,17 +17,11 @@ @implementation AppDelegate
...
-  #ifdef DEBUG
-    jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
-  #else
-    jsCodeLocation = [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
-  #endif
-RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:jsCodeLocation
-                                                      moduleName:@"emurgo"
-                                               initialProperties:nil
-                                                   launchOptions:launchOptions];
+  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];
```
Comment (nicarq) [inline ios/emurgo/AppDelegate.m]: do you know how this knows if this is a local bundled deployment (DEBUG) or a file is embedded (RELEASE)?

Comment (nicarq) [reply ios/emurgo/AppDelegate.m]: nvm

File: package.json [hunk 4]
```diff
@@ -18,9 +18,10 @@
   },
   "dependencies": {
     "@babel/cli": "7.2.3",
-    "@babel/core": "7.4.0",
+    "@babel/core": "^7.4.3",
```
Comment (nicarq) [inline package.json]: I would use specific versions, so no `^`

File: package.json [hunk 5]
```diff
@@ -47,13 +48,13 @@
...
-    "react": "16.5.0",
+    "react": "16.8.3",
     "react-intl": "^2.8.0",
     "react-markdown": "^4.0.6",
-    "react-native": "0.57.1",
+    "react-native": "0.59.3",
     "react-native-background-timer": "^2.1.1",
     "react-native-camera": "^1.3.0",
-    "react-native-cardano": "https://github.com/crossroadlabs/react-native-cardano.git",
+    "react-native-cardano": "https://github.com/Emurgo/react-native-cardano.git#upgrade-to-rn-059",
```
Comment (nicarq) [inline package.json]: I would try to merge that one before this one, so this PR can have the right path

File: package.json [hunk 6]
```diff
@@ -47,13 +48,13 @@
     "node-fetch": "^2.2.0",
     "p-limit": "^2.0.0",
     "path-browserify": "^1.0.0",
-    "react": "16.5.0",
+    "react": "16.8.3",
     "react-intl": "^2.8.0",
     "react-markdown": "^4.0.6",
-    "react-native": "0.57.1",
+    "react-native": "0.59.3",
```
Comment (nicarq) [inline package.json]: I thought we were going to upgrade up to `0.59.10`

File: package.json [hunk 7]
```diff
@@ -83,20 +84,20 @@
...
-    "babel-jest": "23.6.0",
+    "babel-jest": "24.7.1",
     "babel-plugin-react-intl": "^3.0.1",
     "babel-plugin-require-context-hook": "^1.0.0",
     "eslint-plugin-flowtype": "^3.2.0",
     "extract-react-intl-messages": "^0.14.0",
     "flow-bin": "0.78",
-    "jest": "23.6.0",
-    "metro-react-native-babel-preset": "^0.51.1",
+    "jest": "^24.7.1",
```
Comment (nicarq) [inline package.json]: fix version plz 😄

**Comment by v-almonacid** [comment]
Closing as we are going to upgrade from another branch. All review comments will be addressed there.
